OKTA-537418 ISV submission prerequisite (#3827)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/submit-app-prereq/main/index.md

---
title: OIN submission requirements
meta:
  - name: description
    content: Use this guide to prepare all the requirements prior to submitting your integration to Okta for publication in the Okta Integration Network.
layout: Guides
---

This guide provides you with a list of requirements you need to prepare prior to submitting your integration for publication in the [Okta Integration Network (OIN)](https://www.okta.com/integrations/).

---

**Learning outcome**

> * Understand the requirements necessary for the submission process using the [OIN Manager](https://oinmanager.okta.com).

Reviewer: Maybe...

"... necessary to submit your integration using the OIN Manager..."?

**What you need**

> * a built and tested OIN integration that is ready for Okta verification

Reviewer: could use "that's" for more informal feel

---

## Overview

Prior to using the [OIN Manager](https://oinmanager.okta.com)&nbsp;to submit your OIN integration, you need to review the following guidelines to prepare for the artifacts requested during the submission process:

* [Logo guidelines](#logo-guidelines)
* [App description guidelines](#app-description-guidelines)

> * [Use case guidelines](#use-case-guidelines)

Author: I added a new section. Please review.

* [Test account guidelines](#test-account-guidelines)
* [Customer configuration document guidelines](#customer-configuration-document-guidelines)

Before you submit your integration, ensure that your integration doesn't use unsupported OIN Okta features. See [OIN limitations](#oin-limitations).

## Logo guidelines

The app logo that you submit to the OIN Manager in the **General Settings** > **App icon** field must conform to the following guidelines:

* The logo file size must be less than one MB.
* The logo dimension must be square (1:1 aspect ratio).

>    > **Note:** OIN app icons are placed in a 200 x 200 pixels grid.

Reviewer: I think either:
- ... are placed in a grid that's 200 x 200 pixels
- ... are placed in a 200 x 200 pixel grid

* Submit an icon rather than a wordmark (a logo that includes the company or product name).

   > **Note:** The company name is already listed in plain text in the OIN catalog, so adding a company name image is redundant.

* Don't use logos with the trademark (:tm:) symbol.
* The recommended logo is an image in PNG format with a transparent background.
    * GIF and JPEG/JPG formats are also acceptable
    * A colored background is acceptable if it's a part of the logo color scheme

### Logo tips

* If your logo isn't in one of the acceptable formats, ask your design team to convert your existing logo to the preferred PNG format. If you have difficulty with converting your image to an acceptable format, contact the [OIN team](mailto:oin@okta.com).

* If your logo isn't square, consider using your website favicon. Alternatively, you can use the first letter of your app wordmark and convert it to a square image.

> * You can use a slightly non-square icon, but take into account the scaling that occurs when your logo is fitted into a 200 x 200 pixels space. If you have additional logo questions, contact the [OIN team](mailto:oin@okta.com).

Reviewer: "is fitted..." -> "is resized to fit a 200 x 200 pixel area."

## Use case guidelines

Integrations are organized by use cases in the OIN catalog. From the OIN Manager, you can specify the use case categories that your integration supports in the **General Settings** > **App use case** field. You can select up to five use cases. Use the following description list to determine the appropriate use case category for your integration:

| Use case | Integration capability |
| -------- | ---------------------- |
| [Single Sign-On](https://www.okta.com/integrations/?category=single-sign-on) (most common) | Enables users to access your application from any device with a single entry of their Okta user credentials. This use case is appropriate for SAML and OpenID Connect (OIDC) integrations. |
| [Automation](https://www.okta.com/integrations/?category=automation)  | Automates business processes and Okta administration tasks. Most integrations in this use case are API service integrations that access Okta’s APIs using OAuth 2.0.  |

> | [Centralized Logging](https://www.okta.com/integrations/?category=centralized-logging) | Aggregates Okta logs into a central location, like a Security Information and Event Management (SIEM) tool, for optimized searching and alerting capabilities. Integrations in this use case are API service integrations that poll the Okta API for System Logs using OAuth 2.0. |

Reviewer: I'm not sure if it would be confusing to people from whom English isn't their first language, but what if:

"Integrations that fall under this use case are API ..."
otherwise maybe:
"API service integrations that poll the Okta API for System Logs using OAuth 2.0 support this use case." (or similar)

Author: I'll use the latter suggestion. Thanks!

| [Directory and HR Sync](https://www.okta.com/integrations/?category=directory-and-hr-sync) | Provides synchronization capabilities for external-sourced user profiles with the Okta Universal Directory. This use case is most common for HR solutions using SCIM or Workflow Connectors. |

> | [Lifecycle Management](https://www.okta.com/integrations/?category=lifecycle-management) | Enables organizations to securely manage their entire identity lifecycle: from on-boarding to off-boarding, and ensuring compliance requirements are met as user roles evolve and access levels change.  This use case is most common with either SCIM or Workflow Connector integrations. |

Reviewer: You only need one space after the period before "This use case..."

| [Identity Proofing](https://www.okta.com/integrations/?category=identity-proofing) | Enables user self-verification to improve identity assurance and approve access for authorized individuals using document-based and/or knowledge-based proofs |

> | [Identity Governance and Administration (IGA)](https://www.okta.com/integrations/?category=identity-governance-and-administration-iga) | Provides the right people access to the right tools at the right time |

Author: These descriptions were provided by PM/PMM, so perhaps @johnbaldo-okta, @tovamarkowitz-okta , and @deniselu can weigh in here. What about using these borrowed words from Product docs?
"Simplifies and manages an organization's identity and access lifecycles across multiple systems"

> | [Identity Governance and Administration (IGA)](https://www.okta.com/integrations/?category=identity-governance-and-administration-iga) | Provides the right people access to the right tools at the right time |

Reviewer: Are all the leftists left out, leaving them left alone without access? :)

> | [Zero Trust](https://www.okta.com/integrations/?category=zero-trust) | Enables secure access for users regardless of their location, device, or network |

Reviewer: Add period to end to be consistent with other descriptions.

Author: Yes, this took me awhile to get used to too. Our style is to not put periods at the end of fragments, unless followed by another sentence.

| [Bot or Fraud Detection](https://www.okta.com/integrations/?category=bot-or-fraud-detection) | Provides protection from inauthentic, forged, or otherwise fraudulent attempts to register, sign-in, recover, or perform identity-related transactions. Most integrations in this use case are API service integrations that send risk signals to Okta using OAuth 2.0. |

> | [Multi-factor Authentication (MFA)](https://www.okta.com/integrations/?category=multi-factor-authentication-mfa) | Provides an additional layer of security with multifactor authentication for an organization’s cloud, mobile, and web apps |

Reviewer: You have directional apostrophes (') - you might want to check if you can disable directional/smart quotes in the editor you're using, unless we want to be using them on the site, which may be the case. 

Also, add a period to end of description.

Author: I changed the directional apostrophe, but left the period out because it's a fragment.

> | [Risk Signal Sharing](https://www.okta.com/integrations/?category=risk-signal-sharing)| Provides enriched context on clients, apps, users, and other first-party subjects to augment and inform Okta's adaptive authentication and authorization decisions |

Reviewer: Add period

### Configuration guide examples

OIDC:

* [Cerby](https://docs.google.com/document/d/e/2PACX-1vRiswyNyRPVYfEMEwPsbMO8Qn11BjAO-FfUsWBBit_IYe88tzQCHTmMPtmF8uPV044HmXQR13adj3LO/pub)
* [Upwave](https://help.upwave.io/en/articles/4129778-okta-configuration-guide)

SAML:

* [GitHub Enterprise](https://saml-doc.okta.com/SAML_Docs/How-to-Configure-SAML-2.0-for-Github_Enterprise.html)
* [Runscope](https://saml-doc.okta.com/SAML_Docs/How-to-Configure-SAML-2.0-for-Runscope.html)
* [Salesforce](https://saml-doc.okta.com/SAML_Docs/How-to-Configure-SAML-2.0-in-Salesforce.html)
* [Zoom.us](https://saml-doc.okta.com/SAML_Docs/How-to-Configure-SAML-2.0-for-Zoom.us.html)

SCIM:

* [Atlassian Cloud](https://confluence.atlassian.com/cloud/configure-user-provisioning-with-okta-957492956.html)
* [Contentful](https://www.contentful.com/help/okta-user-provisioning-integration-with-scim/)
* [Zscaler](https://help.zscaler.com/zia/saml-scim-configuration-guide-okta)

## OIN limitations

Integrations with the following Okta features can't be published in the OIN catalog:

> * **SWA apps:** SWA integrations are no longer accepted for publication in the OIN catalog. However, existing SWA integrations are still maintained by the OIN team.

Reviewer 2: I would also note to contact the OIN team if there needs to be update on their SWA app since they can no longer self-serve in OIN manager

Author: Added a sentence for this info. Please review.

* **Custom authorization server:** To support the potentially large numbers of Okta orgs accessing it through the OIN, an OIDC integration can't use a [custom authorization server](/docs/concepts/auth-servers/#custom-authorization-server), including the default server. You can only use the [Org authorization server](/docs/concepts/auth-servers/#org-authorization-server).

* **Refresh token:**  Refresh tokens aren't supported for apps published in the OIN.

* **Unsupported scopes:** <br>

>    * `offline_access` scope isn't available since refresh tokens aren't supported for apps published in the OIN

Reviewer: Should these two points end in periods?

---

**Learning outcome**

* Understand the requirements necessary for the submission process using the OIN Manager.

**What you need**

* a built and tested OIN integration that is ready for Okta verification

---

## Overview

Prior to using the [OIN Manager](https://oinmanager.okta.com) to submit your OIN integration, you need to review the following guidelines to prepare for the artifacts requested during the submission process:

* [Logo guidelines](#logo-guidelines)
* [App description guidelines](#app-description-guidelines)
* [Use case guidelines](#use-case-guidelines)
* [Test account guidelines](#test-account-guidelines)
* [Customer configuration document guidelines](#customer-configuration-document-guidelines)

Before you submit your integration, ensure that your integration doesn't use unsupported OIN Okta features. See [OIN limitations](#oin-limitations).

## Logo guidelines

The app logo that you submit to the OIN Manager in the **General Settings** > **App icon** field must conform to the following guidelines:

* The logo file size must be less than one MB.

> * The recommended logo dimension is square (1:1 aspect ratio).

Reviewer 3: I think we should frame as requirements and not recommendations

Author: ok, will do.

   > **Note:** OIN app icons are placed in a 200 x 200 pixels grid.

* The recommended logo contains only a logomark and not a logotype (such as a company icon and not the company name).

   > **Note:** The company name is already listed in plain text in the OIN catalog, so adding a company name image is redundant.

> * Don't use logos with both an icon and a wordmark.

Reviewer 3: Still not sure what the difference is between this one and the one above

Author: right, i'll collapse the 2 points.

Author: actually, i've updated this to 2 points:
```
* Don't use logos with both an icon and a wordmark.
* Don't use just a wordmark logo (such as the company name).
```
Let me know if this is ok?

Reviewer 3: Feel like I'm nit picking at this point, but isn't it basically:
* Submit an icon rather than a wordmark / logo which includes the company or product name

Author: No worries about the nit-picking ;-)
I just separated the two points because in the original Logo guideline handout, these points were emphasized so I thought the OIN Ops team wanted to do the same in this requirements doc.
I'll use your suggestion.

* Don't use logos with the trademark (:tm:) symbol.
* The recommended logo is an image in PNG format with a transparent background.
    * GIF and JPEG/JPG formats are also acceptable
    * A colored background is acceptable if it's a part of the logo color scheme

### Logo tips

If your logo doesn't meet the OIN requirements, the following are suggestions for you to conform to the guidelines:

* If your logo isn't in one of the acceptable formats, ask your design team to convert your existing logo to the preferred PNG format. If you have difficulty with converting your image to an acceptable format, contact the OIN team.

* If your logo isn't square, consider using your website favicon. Alternatively, you can use the first letter of your app wordmark and convert it to a square image.

> * You can use a slightly non-square icon, but take into account the scaling that occurs when your logo is fitted into a 200 x 200 pixels space. If you have additional logo questions, contact the [OIN team](mailto:oin@okta.com).

Reviewer 3: remove the first sentence?

Author: Ok, I'll remove the first sentence in this section, it's redundant.

| [Apps for Good](https://www.okta.com/integrations/?category=apps-for-good) | Pre-built integrations with leading apps that enable your workforce to give back |

## Test account guidelines

You need to create a test account on your app so that the OIN team can use it to test and verify your integration. The test account details and credentials are specified in the **General Settings** > **Test Account** fields from the OIN Manager.

In general, the test account allows the OIN team to verify that your integration flow works as expected for your use case. The test account is typically an admin user in your app with additional privileges depending on your use case:

* For a lifecycle management integration, ensure that your admin test account has HR admin privileges to onboard, change roles, or offboard employees on your app.
* For an SSO or SCIM integration, ensure your admin test account has privileges to configure SSO and SCIM (even if your app is already configured with SSO or SCIM). The OIN team needs to verify whether users and/or groups were created by SCIM provisioning or by SAML/OIDC (JiT) on your application.

> * For an API service integration, ensure that your admin test account has privileges configure system settings and to trigger API processes on your app.

Reviewer 3: ```suggestion
* For an API service integration, ensure that your admin test account has privileges configure an API integration and trigger API requests in your application.
```

---
title: OIN submission requirements
meta:
  - name: description
    content: Use this guide to prepare all the requirements prior to submitting your integration to Okta for publication in the Okta Integration Network.
layout: Guides
---

This guide provides you with a list of requirements you need to prepare prior to submitting your integration for publication in the [Okta Integration Network (OIN)](https://www.okta.com/integrations/).

---

> 

Author: Good catch. I'll remove the bullet.

**Learning outcome**

* Understand the requirements necessary to submit your integration using the [OIN Manager](https://oinmanager.okta.com).

**What you need**

* a built and tested OIN integration that's ready for Okta verification

---

## Overview

Prior to using the [OIN Manager](https://oinmanager.okta.com)&nbsp;to submit your OIN integration, you need to review the following guidelines to prepare for the artifacts requested during the submission process:

* [Logo guidelines](#logo-guidelines)
* [App description guidelines](#app-description-guidelines)
* [Use case guidelines](#use-case-guidelines)
* [Customer support contact guidelines](#customer-support-contact-guidelines)
* [Test account guidelines](#test-account-guidelines)
* [Customer configuration document guidelines](#customer-configuration-document-guidelines)

Before you submit your integration, ensure that your integration doesn't use unsupported OIN Okta features. See [OIN limitations](#oin-limitations).

## Logo guidelines

The app logo that you submit to the OIN Manager in the **General Settings** > **App icon** field must conform to the following guidelines:

> * The logo file size must be less than one MB.

Author: Our style guide is to spell out whole numbers right to nine. https://oktawiki.atlassian.net/wiki/spaces/DOC/pages/722929640/InfoDev+Style+Guide

> * The logo file size must be less than one MB.

Reviewer 4: I did see that, but the [MS style guide](https://learn.microsoft.com/en-us/style-guide/numbers) indicates using numerals for measurements and we use numerals elsewhere ([example](https://developer.okta.com/docs/reference/api/apps/#application-logo-operations)).

> * The logo file size must be less than one MB.

Author: You have a valid point here. I'll defer this to our style guru, @susanharper-okta  😄 .

* The logo dimension must be square (1:1 aspect ratio).

>    > **Note:** OIN app icons are placed in a 200 x 200 pixel grid.

Author: This requirement was provided by OIN ops team and verified by DAX team. Please let me know which UI mentions 420x120px, we can resolve this with the dev teams.

>    > **Note:** OIN app icons are placed in a 200 x 200 pixel grid.

Reviewer 4: I found it within Applications > Applications > Create App Integration. Then clicking the upload button for the logo.

![image](https://user-images.githubusercontent.com/89145748/211387555-2a58a731-f400-465d-9e2f-73469f916862.png)

Perhaps I was just looking at the wrong place?

>    > **Note:** OIN app icons are placed in a 200 x 200 pixel grid.

Author: Yes, these instructions are for the OIN Manager, not the Admin Console.

>    > **Note:** OIN app icons are placed in a 200 x 200 pixel grid.

Reviewer 4: That makes sense.

> * Submit an icon rather than a wordmark (a logo that includes the company or product name).

Author: I need to use active language. What about adjusting your suggestion to this?
`- The logo needs to be an icon rather than a wordmark (a graphic that includes the company or product name).`

> * Submit an icon rather than a wordmark (a logo that includes the company or product name).

Reviewer 4: Sounds good.

> * Submit an icon rather than a wordmark (a logo that includes the company or product name).

Author: Sorry, after reviewing all the comments on this section (it's been revised so many times, I lost count. LOL), I'm going to change this list to imperatives. Thanks for your thoughtful comments 👍

   > **Note:** The company name is already listed in plain text in the OIN catalog, so adding a company name image is redundant.

> * Don't use logos with the trademark (:tm:) symbol.

Author: What about 
`* Logos must not have the trademark (:tm:) symbol.`

> * Don't use logos with the trademark (:tm:) symbol.

Reviewer 4: Sounds good to me.

> * The recommended logo is an image in PNG format with a transparent background.

Author: Not sure about this suggestion, as the Ops team want to stress PNG and transparent background recommendation. Not sure if having a **Note** is a strong enough recommendation. I'll leave this up to Ops team.

* If your logo isn't square, consider using your website favicon. Alternatively, you can use the first letter of your app wordmark and convert it to a square image.

* You can use a slightly non-square icon, but take into account the scaling that occurs when your logo is resized to fit a 200 x 200 pixel area. If you have additional logo questions, contact the [OIN team](mailto:oin@okta.com).

<div class="full">

![OIN acceptable logos](/img/oin/oin-logo-size-guide.png)

</div>

### Examples of acceptable logos

<div class="full">

![OIN acceptable logos](/img/oin/oin-logo-guide.png)

</div>

## App description guidelines

The content in the OIN Manager **General Settings** > **App description** field appears in your Integration detail page (under the **Overview** heading) when your integration is published in the OIN catalog. See the [OIN catalog](https://www.okta.com/integrations/) &nbsp;for examples of existing descriptions in their associated Integration detail page.

> In the **App description** field, describe the value that you provide to customers by partnering with Okta. Your description needs to be less than 1024 characters. See a description example in this [ServiceNow Integration detail](https://www.okta.com/integrations/servicenow-ud/) &nbsp;page.

Author: I like your rewording, but have to leave the "provide to customers by partnering with Okta" part (it's from PM). And PM wants a good example, which ServiceNow provides. I'll remove the ServiceNow name and just reference it as "an example".

In general, the test account allows the OIN team to verify that your integration flow works as expected for your use case. The test account is typically an admin user in your app with additional privileges depending on your use case:

* For a lifecycle management integration, ensure that your admin test account has HR admin privileges to onboard, change roles, or offboard employees on your app.
* For an SSO or SCIM integration, ensure your admin test account has privileges to configure SSO and SCIM (even if your app is already configured with SSO or SCIM). The OIN team needs to verify whether users and/or groups were created by SCIM provisioning or by SAML/OIDC (JiT) on your application.
* For an API service integration, ensure that your admin test account has privileges to configure an API integration and trigger API requests in your application.

> **Note:** The OIN team recommends `isvtest@okta.com` as the test account username, however, you can provide an alternative username with a different domain.

## Customer configuration document guidelines

A configuration guide helps your customers understand how to configure your Okta integration to work with your cloud application.

> You need to provide a configuration guide as part of the OIN submission process. Your guide is provided to administrators through the Okta Admin Console. Okta checks your document for general adherence to the configuration guide instructions. After your integration is published to the OIN, you can make the link public or customer‐accessible.

Author: Thanks for your suggestion. I reworded this part. Please review.

> You need to provide a configuration guide as part of the OIN submission process. Your guide is provided to administrators through the Okta Admin Console. Okta checks your document for general adherence to the configuration guide instructions. After your integration is published to the OIN, you can make the link public or customer‐accessible.

Reviewer 4: Good.

> >**Note:** A configuration guide is required for SCIM and OIDC integrations. It's optional for SAML integrations, as Okta supplies its own documentation with the apps.

Author: I updated this note. There are more integrations than just SCIM and OIDC that require doc. Please review

> >**Note:** A configuration guide is required for SCIM and OIDC integrations. It's optional for SAML integrations, as Okta supplies its own documentation with the apps.

Reviewer 4: Sounds good.

You can create the guide in whatever format works best for you (a webpage, a Google doc, or a PDF are all acceptable), but you need to provide an accessible link to the doc for verification.

### Configuration guide content

The following are section suggestions for your configuration guide:

> 

Author: It will probably look better as a table, but we loose the heading anchors in that format. This content existed for the last 2 years at least, so I'm trying not to change section headers too much.

> 

Reviewer 4: Ah, that makes more sense.

* [Prerequisites](#prerequisites)
* [Supported features](#supported-features)
* [Procedure](#procedure)
* [Troubleshooting tips](#troubleshooting-and-tips)

#### Prerequisites

In this section, specify any prerequisites required before your customer configures your integration in Okta. Examples may include enabling specific Okta features, enabling API access to your SCIM server, or adding a particular version of an integration in Okta.

For example:

> When using SAML as the SSO mode with provisioning, you need to enable a specific account plan on the application side for silent activation.

Author: The blockquotes does not wrap the text. Not sure if there is a pre-wrap option in our env?
We can also use the Note block, but that is a new style convention that I should confirm with the style gurus.

* SP-initiated SSO
* SLO (Single Log Out)
* Force authentication
* Create users
* Update user attributes
* Deactivate users
* Import users
* Import groups
* Sync password
* Profile sourcing

Also include any restrictions. For example:

Okta can't update user attributes for Admin users. This is an API limitation.

> **Note:** You can briefly describe what each feature does. See the SCIM guides from [Configuration guide examples](#configuration-guide-examples).

#### Procedure

This section constitutes the majority of your guide and explains all the configuration steps needed to get your customers set up with your integration. Detail all settings and include any screenshots that can assist the user.

Also include any best practices for your procedure, such as SCIM guidance on setting mappings for attributes, especially required attributes that don't have a default mapping. For example:

> **Note:** The External ID is a required attribute, but it doesn't have a default mapping. This is because some customers prefer to set it to `EmployeeNumber`, and others like to set it to `emailAddress`. Assign the mapping to the correct value for your organization.

#### Troubleshooting and tips

> Include this section if there are known issues that apply to the entire configuration. You can include best practices with the step-by-step procedure instructions. You can also include information on how to contact your organization if the customer has any support queries.

Author: It could be the same or different. The contact mentioned earlier goes in the Integration detail page in the catalog.

* [Contentful](https://www.contentful.com/help/okta-user-provisioning-integration-with-scim/)
* [Zscaler](https://help.zscaler.com/zia/saml-scim-configuration-guide-okta)

## OIN limitations

Integrations with the following Okta features can't be published in the OIN catalog:

* **SWA apps:** Secure Web Authentication (SWA) integrations are no longer accepted for publication in the OIN catalog. However, existing SWA integrations are still maintained by the OIN team. If you need to update your existing SWA integration, contact the [OIN team](mailto:oin@okta.com).

* **Custom authorization server:** To support the potentially large numbers of Okta orgs accessing it through the OIN, an OIDC integration can't use a [custom authorization server](/docs/concepts/auth-servers/#custom-authorization-server), including the default server. You can only use the [Org authorization server](/docs/concepts/auth-servers/#org-authorization-server).

* **Refresh token:**  Refresh tokens aren't supported for apps published in the OIN.

* **Unsupported scopes:** <br>
   * `offline_access` scope isn't available since refresh tokens aren't supported for apps published in the OIN.

>    * `groups` claims (as described in the supported [Scopes](/docs/reference/api/oidc/#scopes) section) isn't supported.

Author: This is the original note. I haven't confirmed with dev or ops team to change this section. I believe that `groups` is both a claim and a scope. I can update this in the next iteration after discussions with devs. 🙇‍♀️

>    * `groups` claims (as described in the supported [Scopes](/docs/reference/api/oidc/#scopes) section) isn't supported.

Reviewer 4: Makes sense. I also noticed "isn't supported" later in the sentence. Not sure if "_aren't supported_" is a better choice or not.

---
title: OIN submission requirements
meta:
  - name: description
    content: Use this guide to prepare all the requirements prior to submitting your integration to Okta for publication in the Okta Integration Network.
layout: Guides
---

> This guide provides you with a list of requirements you need prepare prior to submitting your integration for publication in the [Okta Integration Network (OIN)](https://www.okta.com/integrations/).

Reviewer 3: what does this line mean?

---

**Learning outcome**

* Understand the requirements necessary for the submission process using the OIN Manager.

**What you need**

* a built and tested OIN integration that is ready for Okta verification

---

## Overview

Prior to using the [OIN Manager](https://oinmanager.okta.com) to submit your OIN integration, you need to review the following guidelines to prepare for the artifacts requested during the submission process:

* [Logo guidelines](#logo-guidelines)
* [App description guidelines](#app-description-guidelines)
* [Use case guidelines](#use-case-guidelines)
* [Test account guidelines](#test-account-guidelines)
* [Customer configuration document guidelines](#customer-configuration-document-guidelines)

> Before you follow the submission guidelines, ensure that your integration does not contain [unsupported OIN features](#unsupported-oin-features).

Reviewer 3: what does this line mean?

Reviewer 3: oh sorry, it's right below 😆 

I personally don't think this belongs in this document. Glancing at it, all of them are SSO. All except SWA are OIDC related. I think we could put this in SSO/OIDC build documentation and more as a FAQ / disclaimers in context or as a footnote.

Author: Yes, we should also add it in the OIN SSO documents. One of the complaints from dev-support is that this information is hidden in the various OIN documents and we need to inform potential builders of these limitations early on in the process. I'll add it to the bottom of this page for now and make it a bulleted list (instead of box notes). We can move this content to other pages as we work on them.

> **Refresh token:**  The `refresh_token` option isn't supported for apps published in OIN.

> **Unsupported scopes:** <br>
> * `offline_access` scope isn't available since refresh tokens aren't supported for apps published in OIN.
> * `groups` claims as described in the supported [Scopes](/docs/reference/api/oidc/#scopes) section isn't supported.

<ApiAmProdWarning />

## Logo guidelines

The app logo that you submit to the OIN Manager in the **General Settings** > **App icon** field must conform to the following guidelines:

* The logo file size must be less than one MB.
* The recommended logo dimension has a 1:1 (square) aspect ratio.

   > **Note:** OIN app icons are placed in a 200 x 200 pixels grid.

* The recommended logo contains only a logomark and not a logotype (such as a company icon and not the company name).

   > **Note:** The company name is already listed in plain text in the OIN catalog, so adding a company name image is redundant.

> * Avoid logos with both an icon and a wordmark.

Reviewer 3: This is the same as the above?

I think we should remove "recommended" / "avoid" and make them requirements.

* The logo must be a square icon (200 x 200 recommended) and not contain the company name
> **Note:** The name of the application will be displayed alongside the icon

Reviewer 3: What if we put some screenshots or links to places where the logo is displayed so they can see it in context?

Reviewer 3: Sorry, I just read your Jira comment after going through all this where you suggest the same.

* Avoid logos with the trademark (:tm:) symbol.
* The recommended logo is an image in PNG format with a transparent background.
    * GIF and JPEG/JPG formats are also acceptable
    * A colored background is acceptable if it's a part of the logo color scheme

> ### Comply with the logo guidelines

Reviewer 3: Tips?

If your logo doesn't meet the OIN requirements, the following are suggestions for you to conform to the guidelines:

* If your logo isn't in one of the acceptable formats, ask your design team to convert your existing logo to the preferred PNG format. If you have difficulty with converting your image to an acceptable format, contact the OIN team.

* If your logo isn't square, consider using your website favicon. Alternatively, you can use the first letter of your app wordmark and convert it to a square image.

* You can use a slightly non-square icon, but take into account the scaling that occurs when your logo is fitted into a 200 x 200 pixels space. If you have additional logo questions, contact the OIN team.

<div class="full">

![OIN acceptable logos](/img/oin/oin-logo-size-guide.png)

</div>

### Example of acceptable logos

<div class="full">

![OIN acceptable logos](/img/oin/oin-logo-guide.png)

</div>

## App description guidelines

> An app description is displayed on your integration tile in the OIN catalog. You can specify your app description in the **General Settings** > **App description** field from the OIN Manager. The app description content appears in your Integration detail page, under the **Overview** header in the OIN catalog. See the [OIN catalog](https://www.okta.com/integrations/) for examples of existing app descriptions in their associated Integration page.

Reviewer 3: Some of the descriptions / overviews in the catalog are pretty fancy and not something we can accept from ISVs today. Is there a specific app we could link to so they can get a sense of what they are getting (and an example of a well-written description)

Author: @johnbaldo-okta Can you let me know what is a good example? I'm adding a link to the ServiceNow Integration details page as an example for now : https://www.okta.com/integrations/servicenow-ud/.

Reviewer 3: Thanks, I think that works for now.

Prepare an app description that describes what your integration offers and the supported features. You can also provide value-added messaging for your integration. Your description needs to be less than 1024 characters.

> ## Use case guidelines

Reviewer 3: @tovamarkowitz-okta, side note: now that this is use case and not category of SaaS app, I think we could stop asking for it in the OIN Manager? API Service integrations might be an exception but maybe we can work through that with some other solution. 99% are SSO or provisioning and we should know that based on the type of integration they build.

| [Bot or Fraud Detection](https://www.okta.com/integrations/?category=bot-or-fraud-detection) | Provides protection from inauthentic, forged, or otherwise fradulent attempts to register, sign-in, recover, or perform identity-related transactions. Most integrations in this use case are API service integrations that send risk signals to Okta using OAuth 2.0. |
| [Multi-factor Authentication (MFA)](https://www.okta.com/integrations/?category=multi-factor-authentication-mfa) | Provides an additional layer of security with multifactor authentication for an organization’s cloud, mobile, and web apps |
| [Risk Signal Sharing](https://www.okta.com/integrations/?category=risk-signal-sharing)| Provides enriched context on clients, apps, users, and other first-party subjects to augment and inform Okta's adaptive authentication and authorization decisions |
| [Apps for Good](https://www.okta.com/integrations/?category=apps-for-good) | Pre-built integrations with leading apps that enable your workforce to give back |

## Test account guidelines

> You need to create a test account on your app so that the OIN team can use it to test and verify your integration. The test account details and credentials are specified  in the **General Settings** > **Test Account** fields from the OIN Manager.

Reviewer 3: Maybe we can work in the high level (several statements that start with "In general" below) and mention that we're trying to test the functionality and customer configuration.

Then list out bulleted requirements?
* must be an admin (in your application, not your Okta org) with the correct privileges
* should use the email isvtest@okta.com
* ...etc

??? Work on:

We need a test user with admin access who has the rights to configure SSO for SAML/SCIM. Even if your support team has configured SSO or SCIM, we need access to verify if users and/or groups on the application end are created by SCIM Provisioning or SAML/OIDC (JiT).

In general, the test account allows the OIN team to verify that your integration flow works as expected for your use case.

We prefer isvtest@okta.com to be a test user. If you can't use this email to create a test user, you can provide a specific account in a different domain.

??? Perhaps remove/merge with the above ???

Depending on the use case of our app, the permissions and roles for your test account are varied. For example:

> * For an SSO integration, you can provide a regular, non-admin user test account for the OIN team to test Single Sign-On with your app.

Reviewer 3: Question for Abdullah, but can we just ask for an admin user since you can also test SSO for an admin user?

If we want to remove the nuance for each type can we say that the user should have permission to configure the integration?

Author: I've updated this section to incorporate @mohammadabdullah-okta's feedback on user permissions.

* For an API service integration, you can provide a test account with privileges to trigger an API process on your app.

> **Note:** Ensure that proper scopes are defined for OAuth 2.0 authentication.

In general, the test account provides the OIN team with the ability to verify that your integration flow works as expected for your use case. Provide the necessary permissions required for your app. If you have further questions, contact the OIN team.

## Customer configuration document guidelines

A configuration guide helps your customers understand how to configure your Okta integration to work with your cloud application.

You need to provide a configuration guide as part of the OIN submission process. Your guide is provided to administrators through the Okta Admin Console. Okta checks your document for general adherence to the configuration guide instructions. After your integration is published to the OIN, you can make the link public or customer‐accessible.

>**Note:** A configuration guide is required for SCIM and OIDC integrations. It's optional for SAML integrations, as Okta supplies its own documentation with the apps.

> You can create the guide in whatever format works best for you: a webpage, a Google or Word doc, or a PDF are all acceptable.

Reviewer 3: @mohammadabdullah-okta , wouldn't it be best to collect the actual link so that you don't need to email and ask for that later?

I wonder if we should be collecting a draft link (optional), published link (required)

Reviewer 5: +1 on link so that there's consistency and reduces back and forth with the ISV

Author: I'll update this sentence to request for a doc link that is publicly accessible

> ### Configuration guide examples

Reviewer 3: @mohammadabdullah-okta, could we have one example for each or if we are providing examples for different cases, can we note which is which? For example: SAML when the administrator needs to contact support, etc.

OIDC:

* [Cerby](https://docs.google.com/document/d/e/2PACX-1vRiswyNyRPVYfEMEwPsbMO8Qn11BjAO-FfUsWBBit_IYe88tzQCHTmMPtmF8uPV044HmXQR13adj3LO/pub)
* [Upwave](https://help.upwave.io/en/articles/4129778-okta-configuration-guide)

SAML:

* [GitHub Enterprise](https://saml-doc.okta.com/SAML_Docs/How-to-Configure-SAML-2.0-for-Github_Enterprise.html)
* [Runscope](https://saml-doc.okta.com/SAML_Docs/How-to-Configure-SAML-2.0-for-Runscope.html)
* [Salesforce](https://saml-doc.okta.com/SAML_Docs/How-to-Configure-SAML-2.0-in-Salesforce.html)
* [Zoom.us](https://saml-doc.okta.com/SAML_Docs/How-to-Configure-SAML-2.0-for-Zoom.us.html)

SCIM:

* [Atlassian Cloud](https://confluence.atlassian.com/cloud/configure-user-provisioning-with-okta-957492956.html)
* [Contentful](https://www.contentful.com/help/okta-user-provisioning-integration-with-scim/)
* [Zscaler](https://help.zscaler.com/zia/saml-scim-configuration-guide-okta)

> ### Configuration guide content

Reviewer 3: Should we put the examples below the guidelines to encourage people to actually look through these?

Author: Sure, I can do that. The doc examples were always presented first in the past (so I thought it was intentional).

The following are section suggestions for your configuration guide:

* [Prerequisites](#prerequisites)
* [Supported features](#supported-features)
* [Procedure](#procedure)
* [Troubleshooting tips](#troubleshooting-and-tips)

#### Prerequisites

In this section, specify any prerequisites required before your customer configures your integration in Okta. Examples may include enabling specific Okta features, enabling API access to your SCIM server, or adding a particular version of an integration in Okta.

For example:

> When using SAML as the SSO mode with provisioning, you need to enable a specific account plan on the application side for silent activation.

Reviewer 3: nice to have examples!

Is this saying that you need a certain account type / payment plan for the feature?

#### Supported features

> In this section of your guide, you want to outline what features your application supports. For example:

Reviewer 3: >you want to outline what features your application supports.

list the features your application supports?

* IdP-initiated SSO
* SP-initiated SSO
* SLO (Single Log Out)
* Force Authentication
* Create Users
* Update User Attributes
* Deactivate Users
* Import Users
* Import Groups
* Sync Password
* Profile Sourcing (formerly called Profile Mastering)

Also include any restrictions. For example:

Okta can't update user attributes for Admin users. This is an API limitation.

> > **Note:** You can briefly describe what each feature does. See the SCIM guides from the earlier example section.

Reviewer 3: Do we have a description somewhere so that people don't need to come up with their own descriptions?

#### Procedure

This section constitutes the majority of your guide and explains all the configuration steps needed to get your customers set up with your integration. Detail all settings and include any screenshots that can assist the user.

> Also include any best practices for your procedure, such as guidance on setting mappings for attributes, especially required attributes that don't have a default mapping. For example:

Reviewer 3: Is this only a SCIM thing? Should we mention that?

> **Note:** The External ID is a required attribute, but it doesn't have a default mapping. This is because some customers prefer to set it to `EmployeeNumber`, and others like to set it to `emailAddress`. Assign the mapping to the correct value for your organization.

#### Troubleshooting and tips

You need to only include this section if there are known issues that apply to the entire configuration. In general, you should include best practices with the step-by-step procedure instructions.

> You may also want to include information on how to contact your organization if the customer has any support queries.

Reviewer 3: Can we have a template for them to get started with? Ideally one, but could be one for each integration type if they are different enough.

Author: I was provided with this SCIM doc instruction template: https://saml-doc.okta.com/Provisioning_Docs/SCIM_Configuration_Guide_Instructions.pdf. @johnbaldo-okta , would you like to add this link as a template?
